### PR TITLE
[Feature Fix] fix icons of components/projects in file grid

### PR DESCRIPTION
**Pull request by leb2dg**
# Purpose

This resolves issue #2783. The icon that is displayed next to projects and components is an open folder.

# Changes

The *_serialize_node()* function of the *NodeFileCollector* class in *rubeus.py* returns the category of the node in addition to the other data. The *_serialize_node()* function of *NodeProjectCollector* class in *rubeus.py* returns the *nodeType* of the node in addition to the other data. The iconmap is required in fangorn.js so that the icons that appear next to the component is the same as the one on the project organizer view. The logic from *projectorganizer.js* is consolidated in *fangorn.js* as *resolveIconHelper()*. This allows the icon map to be used in the file grid and the project organizer without duplicate code.

# Side effects

In the file grid, the category is now returned with the other data (such as nodeType or kind). In the project organizer, the nodeType is now returned with the other data. The file grid displays the same icons as the project organizer.

Fix:
screen shot 2015-05-28 at 1 21 22 pm

Original:
screen shot 2015-05-28 at 1 22 38 pm

### CODE REVIEW

**Comment by lbanner** [comment]
Nice!

File: website/static/js/fangorn.js [hunk 1]
```diff
@@ -182,12 +184,43 @@ var uploadRowTemplate = function(item){
  * @private
  */
 function _fangornResolveIcon(item) {
+    var icons = iconmap.projectIcons;
```
Comment (samanehsan) [inline website/static/js/fangorn.js]: `icons` is the same as `projectIcons` and can be removed.

**Comment by leb2dg** [comment]
@samanehsan I used the same code that is in projectorganizer.js (line 325). I didn't write that one but should it be changed as well?

**Comment by samanehsan** [comment]
Actually, if it's the same code, could you consolidate it into one place? That way we won't need to change it in multiple places if that code has to be updated in the future.

**Comment by leb2dg** [comment]
@samanehsan I consolidated the code, thanks for your help!

**Comment by samanehsan** [comment]
Thank you! @sloria, I think this is good to go. :tada:

File: website/static/js/fangorn.js [hunk 2]
```diff
@@ -184,6 +186,59 @@ var uploadRowTemplate = function(item){
...
+        var template = m('span', { 'class' : iconType});
+        return template;
+    }
+    if (item.data.nodeType === 'smartFolder') {
+        return returnView('smartCollection');
+    }
+    if (item.data.nodeType === 'folder') {
+        return returnView('collection');
+    }
+    if (item.data.nodeType === 'pointer' && !item.parent().data.nodeType === 'folder') {
```
Comment (sloria) [inline website/static/js/fangorn.js]: As found by jshint--the use of `!` is confusing here. Better to just use `!==`.

Comment (sloria) [inline website/static/js/fangorn.js]: Will make this change when I merge.

File: website/static/js/fangorn.js [hunk 3]
```diff
@@ -184,6 +186,59 @@ var uploadRowTemplate = function(item){
 };
 
 /**
+ * Returns custom icons for OSF depending on the type of item. Used for non-file icons.
+ * @param {Object} item A Treebeard _item object. Node information is inside item.data
+ * @this Treebeard.controller
+ * @returns {Object}  Returns a mithril template with the m() function.
+ */
+function resolveIconHelper(item) {
```
Comment (sloria) [inline website/static/js/fangorn.js]: No need to add "Helper" to function names. Use verbs when possible, e.g. `resolveIcon`. 

Will make this change when I merge.

**Comment by leb2dg** [comment]
This PR prevents the private folder icon from being shown, it shows the item icon instead, my next PR for issue #2951 will resolve this since it involves the private folders. @sloria is that okay for should I create another issue & PR?

The problem:
screen shot 2015-06-05 at 9 58 30 am

**Comment by nicipfeiffer** [comment]
@sloria should we defer testing this until the private folder icons have been sorted out?

**Comment by leb2dg** [comment]
@nicipfeiffer I would defer...there are a few changes I need to make with *resolveIconView* to get both the project organizer and the file grid to display the icons correctly. I don't want you guys to waste your time.

**Pull request merged**